root/static/scripts/common/components/EntityLink.js

```diff
@@ -32,96 +36,109 @@ const DeletedLink = ({name, allowNew}) => {
+    <span className={className}>
+      {'('}
+      {isolateText(comment)}
+      {')'}
+    </span>
   </Frag>
 );
 
-class EventDisambiguation extends React.Component {
-  render() {
-    let event = this.props.event;
-    let dates = formatDatePeriod(event);
-    if (!dates && !event.cancelled) {
-      return null;
-    }
-    return (
-      <Frag>
-        {dates ? bracketed(dates) : null}
-        {event.cancelled
-          ? <Comment className="cancelled" comment={l('cancelled')} />
-          : null}
-      </Frag>
-    );
+const EventDisambiguation = ({event}: {|+event: EventT|}) => {
+  const dates = formatDatePeriod(event);
+  if (!dates && !event.cancelled) {
+    return null;
   }
-}
-
-class AreaDisambiguation extends React.Component {
-  render() {
-    let area = this.props.area;
-
-    if (!area.ended) {
-      return null;
-    }
+  return (
+    <Frag>
+      {dates ? bracketed(dates) : null}
+      {event.cancelled
+        ? <Comment className="cancelled" comment={l('cancelled')} />
+        : null}
+    </Frag>
+  );
+};
 
-    let comment;
-    let beginYear = area.begin_date ? area.begin_date.year : null;
-    let endYear = area.end_date ? area.end_date.year : null;
+const AreaDisambiguation = ({area}: {|+area: AreaT|}) => {
+  if (!area.ended) {
+    return null;
+  }
 
-    if (beginYear && endYear) {
-      comment = l('historical, {begin}-{end}', {begin: beginYear, end: endYear});
-    } else if (endYear) {
-      comment = l('historical, until {end}', {end: endYear});
-    } else {
-      comment = l('historical');
-    }
+  let comment;
+  const beginYear = area.begin_date ? area.begin_date.year : null;
+  const endYear = area.end_date ? area.end_date.year : null;
 
-    return <Comment className="historical" comment={comment} />;
+  if (beginYear && endYear) {
+    comment = l('historical, {begin}-{end}', {begin: beginYear, end: endYear});
+  } else if (endYear) {
+    comment = l('historical, until {end}', {end: endYear});
+  } else {
+    comment = l('historical');
   }
-}
+
+  return <Comment className="historical" comment={comment} />;
+};
 
 const NoInfoURL = ({url, allowNew}) => (
   <Frag>
     <a href={url}>{url}</a>
     {' '}
-    <DeletedLink name={'[' + l('info') + ']'} allowNew={allowNew} />
+    <DeletedLink allowNew={allowNew} name={'[' + l('info') + ']'} />
   </Frag>
 );
 
-const EntityLink = (props = {}) => {
-  let {
-    allowNew,
-    content,
-    entity,
-    hover,
-    showDeleted = true,
-    showDisambiguation,
-    subPath,
-    ...anchorProps
-  } = props;
-
+/* eslint-disable sort-keys, flowtype/sort-keys */
+type EntityLinkProps = {
+  +allowNew?: boolean,
+  +content?: React.Node,
+  +entity: CoreEntityT,
+  +hover?: string,
+  +showDeleted?: boolean,
+  +showDisambiguation?: boolean,
+  +subPath?: string,
+
+  // ...anchorProps
+  href?: string,
+  title?: string,
+  +target?: '_blank',
+};
+/* eslint-enable sort-keys, flowtype/sort-keys */
```

Comment:
I wanted the keys that are only supposed to exist in the `anchorProps` spread to be grouped together, so I disabled those eslint rules for this block.